src/common/sourceCodeManipulations.ts

```diff
@@ -0,0 +1,108 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+import { getSyntaxErrorIn } from './sourceUtils';
+import { invalidLogPointSyntax } from '../dap/errors';
+import { ProtocolError } from '../dap/protocolError';
+import ts from 'typescript';
+
+/**
+ * function (params) { code } => function (params) { catchAndReturnErrors(code) }
+ * statement => function () { return catchAndReturnErrors(return statement) }
+ * statement; statement => function () { catchAndReturnErrors(statement; return statement;) }
+ * */
+export function codeToFunctionReturningErrors(
+  parameterNames: string,
+  statements: ReadonlyArray<ts.Statement>,
+) {
+  if (statements.length === 1 && statements[0].kind === ts.SyntaxKind.FunctionDeclaration) {
+    const functionDeclarationCode = statements[0].getText();
+    const callFunctionCode = `return (${functionDeclarationCode}).call(this, ${parameterNames});`;
+    return codeToFunctionExecutingCodeAndReturningErrors(parameterNames, callFunctionCode, true);
+  } else {
+    return statementToFunctionReturningErrors(parameterNames, statements, true);
+  }
+}
+
+/**
+ * code => (parameterNames) => return catchAndReturnErrors(code)
+ * */
+export function codeToFunctionExecutingCodeAndReturningErrors(
```

Comment:
This is only used internally

```suggestion
function codeToFunctionExecutingCodeAndReturningErrors(
```

Reply:
Done